pkg/gitproviders/repo_url.go

```diff
@@ -139,18 +150,33 @@ func normalizeRepoURLString(url string, providerName GitProviderName) string {
 		url = url + ".git"
 	}
 
-	sshPrefix := fmt.Sprintf("git@%s.com:", providerName)
-	httpsPrefix := fmt.Sprintf("https://%s.com/", providerName)
+	u, err := parseGitURL(url)
+	if err != nil {
+		return "", fmt.Errorf("could not parse git repo url while normalizing %q: %w", url, err)
+	}
+
+	return fmt.Sprintf("ssh://git@%s%s", u.Host, u.Path), nil
+}
 
-	if strings.HasPrefix(url, sshPrefix) {
-		trimmed = strings.TrimPrefix(url, sshPrefix)
-	} else if strings.HasPrefix(url, httpsPrefix) {
-		trimmed = strings.TrimPrefix(url, httpsPrefix)
+// ViperGetStringMapString looks up a command line flag or env var in the format "foo=1,bar=2"
+// GetStringMapString tries to JSON decode the env var
+// If that fails (silently), try and decode the classic "foo=1,bar=2" form.
+// https://github.com/spf13/viper/issues/911
+func ViperGetStringMapString(key string) map[string]string {
+	sms := viper.GetStringMapString(key)
+	if len(sms) > 0 {
+		return sms
 	}
 
-	if trimmed != "" {
-		return fmt.Sprintf("ssh://git@%s.com/%s", providerName, trimmed)
+	ss := viper.GetStringSlice(key)
+	out := make(map[string]string, len(ss))
```

Comment:
I am not sure there's any benefit in providing the `cap` here?

Reply:
Nope, brought the code over from https://github.com/spf13/pflag/blob/master/string_to_string.go#L90-L98 which might need an attribution actually?